core/src/main/java/org/apache/iceberg/ManifestFiles.java

```diff
@@ -149,6 +160,33 @@ private ManifestFiles() {
     throw new UnsupportedOperationException("Cannot write manifest for table version: " + formatVersion);
   }
 
+  private static org.apache.avro.Schema getManifestAvroSchema() {
+    return AvroSchemaUtil.convert(ManifestFile.schema(), ImmutableMap.of(
+        ManifestFile.schema().asStruct(), GenericManifestFile.class.getName(),
+        ManifestFile.PARTITION_SUMMARY_TYPE, GenericPartitionFieldSummary.class.getName()
+    ));
+  }
+
+  public static byte[] encode(ManifestFile manifestFile) throws IOException {
+    try (ByteArrayOutputStream out = new ByteArrayOutputStream()) {
+      BinaryEncoder encoder = EncoderFactory.get().binaryEncoder(out, null);
+      DatumWriter<GenericManifestFile> writer = new GenericAvroWriter<>(getManifestAvroSchema());
+      writer.write((GenericManifestFile) manifestFile, encoder);
```

Comment:
hmm. I am not sure if we should tie Flink state serialization to Iceberg serialization

Reply:
I agree with this, but I'm okay making some utils available for this.